documentation/decisions/01 - Using a Compound Component Pattern for Autocomplete.md

```diff
@@ -0,0 +1,28 @@
+# Using a compound component patter to build Autocomplete
+
+## Date
+
+October 3, 2019
+
+## Contributors
+
+- Andrew Musgrave
+- Chloe Rice
+- Dan Rosenthal
+- Daniel Leroux
+- Dominic McPhee
+- Sara Hill
+
+## Summary
+
+There is a need for Polaris to provide smaller more reusable components so that Polaris can be used by everyone without needing to fork component. The re-development of the Autocomplete component provided an opportunity for exploration.
```

Comment:
```suggestion
There is a need for Polaris to provide smaller, more composable components so that Polaris can be extended without needing to fork components. The re-development of the Autocomplete component provided an opportunity for exploration of the compound component pattern.
```

Reply:
This might need one more sentence around why composable components allow greater flexibility. IE: not needing to add a prop for any new configuration, as it can be handled as children. Adding an `Icon` to an `Option` feels like a good example to explain this, since this approach would not require adding/maintaining another prop.